lib/migrator.js

```diff
@@ -6,53 +6,158 @@ const { moveFile, removeDirs } = require('./utils/file')
+  findClassicComponentTemplates() {
+    const templateFolderPath = path.join(this.projectRoot, 'app/templates/components');
+    const templateFilePaths = glob.sync(`${templateFolderPath}/**/*.hbs`);
+
+    return templateFilePaths;
+  }
+
+  findClassicComponentClasses() {
+    const classFolderPath = path.join(this.projectRoot, 'app/components');
+    const classFilePaths = glob.sync(`${classFolderPath}/**/*.js`);
+
+    return classFilePaths;
+  }
+
+  findTemplates() {
+    const templateFolderPath = path.join(this.projectRoot, 'app/templates');
+    const templateFilePaths = glob.sync(`${templateFolderPath}/**/*.hbs`);
+
+    return templateFilePaths;
+  }
+
+  skipTemplatesUsedAsLayoutName(templateFilePaths) {
+    console.info(`\nChecking if any component templates are used as templates of other components using \`layoutName\``);
+
+    const classFilePaths = this.findClassicComponentClasses();
+    const componentsWithLayoutName = getLayoutNameTemplates(classFilePaths);
+
+    if (componentsWithLayoutName.length) {
+      componentsWithLayoutName.sort().forEach(component => {
+        console.info(`❌ Did not move '${component}' due to usage as "layoutName" in a component`);
+      });
+
+      templateFilePaths = templateFilePaths.filter(templateFilePath => {
+        // Extract '/app/templates/components/nested1/nested-component.hbs'
+        const filePathFromApp = templateFilePath.slice(this.projectRoot.length);
+
+        // Extract '/components/nested1/nested-component.hbs'
+        const filePathFromAppTemplates = filePathFromApp.slice('app/templates/'.length);
+
+        // Extract 'components/nested1/nested-component'
+        const classFilePath = filePathFromAppTemplates.slice(1).replace('.hbs', '');
+
+        return !componentsWithLayoutName.includes(classFilePath);
       });
     }
 
-    let sourceTemplatesPath = path.join(this.options.projectRoot, 'app/templates');
-    var sourceTemplateFilePaths = glob.sync(`${sourceTemplatesPath}/**/*.hbs`);
-    let templatesInPartials = getPartialTemplates(sourceTemplateFilePaths);
-    if (templatesInPartials.length) {
-      sourceComponentTemplateFilePaths = sourceComponentTemplateFilePaths.filter(sourceTemplateFilePath => {
-        let sourceTemplatePathInApp = sourceTemplateFilePath.slice(this.options.projectRoot.length); // '/app/templates/components/nested1/nested-component.hbs'
-        if (/\/\-[\w\-]+\.hbs/.test(sourceTemplatePathInApp)) {
-          sourceTemplatePathInApp = sourceTemplatePathInApp.replace('/-', '/');
+    return templateFilePaths;
+  }
+
+  skipTemplatesUsedAsPartial(templateFilePaths) {
+    console.info(`\nChecking if any component templates are used as partials`);
+
+    const componentsWithPartial = getPartialTemplates(this.findTemplates());
+
+    if (componentsWithPartial.length) {
+      componentsWithPartial.sort().forEach(component => {
+        console.info(`❌ Did not move '${component}' due to usage as a "partial"`);
```

Comment:
Should probably say where the `{{partial` invocation was found

Reply:
Can you elaborate on what you'd like the log message to say?

A test run shows that the log message currently looks like:
```
Checking if any component templates are used as partials
❌ Did not move 'components/partials/partials-template' due to usage as a "partial"
```

Reply:
Basically to say the template name (which the message does now) and the name of hte template that _uses_ it as a partial.

> ❌ Did not move 'components/partials/partials-template' due to usage as a "partial" in 'templates/index.hbs'